Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -1647,22 +1648,15 @@ namespace OpenInfraPlatform
 				typename IfcEntityTypesT::IfcBooleanOperand ifcSecondOperand = boolResult->SecondOperand;
 
 				carve::csg::CSG::OP csgOperation = carve::csg::CSG::A_MINUS_B;
-				if (ifcBooleanOperator == typename IfcEntityTypesT::IfcBooleanOperator::ENUM::ENUM_UNION)
-				{
+
+				if (typeid(IfcEntityTypesT::IfcBooleanOperator::ENUM::ENUM_UNION) == typeid(ifcBooleanOperator))
 					csgOperation = carve::csg::CSG::UNION;
-				}
-				else if (ifcBooleanOperator == typename IfcEntityTypesT::IfcBooleanOperator::ENUM::ENUM_INTERSECTION)
-				{
+				else if (typeid(IfcEntityTypesT::IfcBooleanOperator::ENUM::ENUM_INTERSECTION) == typeid(ifcBooleanOperator))
 					csgOperation = carve::csg::CSG::INTERSECTION;
-				}
-				else if (ifcBooleanOperator == typename IfcEntityTypesT::IfcBooleanOperator::ENUM::ENUM_DIFFERENCE)
-				{
+				else if (typeid(IfcEntityTypesT::IfcBooleanOperator::ENUM::ENUM_DIFFERENCE) == typeid(ifcBooleanOperator))
 					csgOperation = carve::csg::CSG::A_MINUS_B;
-				}

```

Comment:
These are really weird; but as long as it works ... :S

Reply:
I wasn't able to find an example file that makes use of  Intersection, Difference or Union, but I can create an issue to check this part with an example file to see if it works properly.